app/forms/statuses/form.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Statuses
+  class Form < ApplicationForm
+    attr_reader :submit_label
+
+    def initialize(submit_label: nil)
+      super()
+
+      @submit_label = submit_label || I18n.t(:button_save)
+    end
+
+    form do |statuses_form|
+      statuses_form.text_field(
+        label: attribute_name(:name),
+        name: :name,
+        required: true,
+        input_width: :medium
+      )
+
+      statuses_form.text_field(
+        label: attribute_name(:default_done_ratio),
+        name: :default_done_ratio,
+        caption: percent_complete_field_caption,
+        required: true,
+        type: :number,
+        min: 0,
+        max: 100,
+        maxlength: 7,
+        autocomplete: "off",
+        input_width: :small
+      )
+
+      statuses_form.check_box(
+        label: attribute_name(:is_closed),
+        name: :is_closed
+      )
+
+      unless already_default_status?
+        statuses_form.check_box(
+          label: attribute_name(:is_default),
+          name: :is_default
+        )
```

Comment:
I personally would prefer to still display the checkbox in a disabled state if the current status is the default one. And there are probably even better ways but this would be an easy improvement IMO.

Reply:
I only ported the existing behavior without questioning it.
When looking at it with Parimal some months ago, he was not satisfied with this way of choosing which attribute is the default, but did not want to touch it for the moment. I'll leave it like this too.

Reply:
We [discussed it](https://community.openproject.org/meetings/3690#item-5949) in frontend daily, and it ended up as a bug: [Bug: Statuses edit page should show all checkboxes and disable those leading to impossible states (#58360)](https://community.openproject.org/projects/openproject/work_packages/58360/activity)